src/scalar/dusk.rs

```diff
@@ -253,6 +254,35 @@ impl Scalar {
         res
     }
 
+    /// Compute a uniformly distributed random scalar.
+    ///
+    /// Because scalars take 255 bits for encoding it is difficult to generate
+    /// random bit-pattern that ensures to encodes a valid scalar.
+    /// Wrapping the values that are higher than [`MODULUS`] results in hitting
+    /// some values more than others, and zeroing out the highest two bits will
+    /// eliminate some values from the possible results.
+    ///
+    /// This function achieves a uniform distribution of scalars by using
+    /// rejection sampling: random bit-patterns are generated until a valid
+    /// scalar is found.
+    /// The function is not constant time but that shouldn't be a concern since
+    /// no information about the scalar can be gained by knowing the time of
+    /// its generation.
+    pub fn uni_random(rng: &mut (impl RngCore + CryptoRng)) -> Self {
```

Comment:
Any reason to not choose the const generic T where T impl RngCore and CryptoRng?

Reply:
none in particular, I have no preference